src/utils/format.ts

```diff
@@ -41,31 +40,58 @@ export function formatAmount(
   amount?: BN,
   amountPrecision = DEFAULT_PRECISION,
   decimals = DEFAULT_DECIMALS,
+  thousandSeparator: boolean = true,
 ): string | null {
   if (!amount) {
     return null
   }
+  const actualDecimals = Math.min(amountPrecision, decimals)
+  const { integerPart, decimalPart } = _decomposeBn(amount, amountPrecision, actualDecimals)
 
-  const { integerPart, decimalPart } = _decomposeBn(amount, amountPrecision, decimals)
+  const integerPartFmt = thousandSeparator ? _formatNumber(integerPart.toString()) : integerPart.toString()
 
   if (decimalPart.isZero()) {
     // Return just the integer part
-    return _formatNumber(integerPart.toString())
+    return integerPartFmt
   } else {
     const decimalFmt = decimalPart
       .toString()
-      .padStart(decimals, '0') // Pad the decimal part with leading zeros
+      .padStart(actualDecimals, '0') // Pad the decimal part with leading zeros
       .replace(/0+$/, '') // Remove the right zeros
 
     // Return the formated integer plus the decimal
-    return _formatNumber(integerPart.toString()) + DECIMALS_SYMBOL + decimalFmt
+    return integerPartFmt + DECIMALS_SYMBOL + decimalFmt
   }
 }
 
-export function formatAmountFull(amount?: BN, amountPrecision = DEFAULT_PRECISION): string | null {
+export function formatAmountFull(
+  amount?: BN,
+  amountPrecision = DEFAULT_PRECISION,
+  thousandSeparator: boolean = true,
+): string | null {
   if (!amount) {
     return null
   }
 
-  return formatAmount(amount, amountPrecision, amountPrecision)
+  return formatAmount(amount, amountPrecision, amountPrecision, thousandSeparator)
+}
+
+export function parseAmount(amountFmt: string, amountPrecision = DEFAULT_PRECISION): BN | null {
+  if (!amountFmt) {
+    return null
+  }
+
+  const groups = /^(\d+)(?:\.(\d+))?$/.exec(amountFmt)
```

Comment:
Cause BN is only for integers. It doesn't suport decimals. Users enter the amounts with decimals, because they don't work in weis. BN.js works in weis.

Maybe I'm overlooking something, because I also thought it was weird there was not more libs out there doing formatting/parsing of BN.js

Let me know if you can think of a better way

Reply:
Ahhh got it. Makes sense now :)